test/aio5gc/msg/nas/nasDispatcher.go

```diff
@@ -27,57 +28,91 @@ func Dispatch(nasPDU *ngapType.NASPDU, ueContext *context.UEContext, fgc *contex
 		var integrityProtected bool
 		msg, integrityProtected, err = tools.Decode(ueContext, payload, false)
 		if !integrityProtected {
-			log.Fatal(errors.New("[5GC][NAS] message integrity could not be verified"))
+			log.Fatal(errors.New("[5GC][NAS] message integrity could not be verified:" + err.Error()))
 		}
 
 	} else {
 		msg, err = tools.DecodePlainNasNoIntegrityCheck(payload)
 	}
-
-	// Hook for changing 5GC behaviour
-	hooks := fgc.GetNasHooks()
```

Comment:
Why not keep it here?
hooks[msg.GmmHeader.GetMessageType()](msg, ueContext, gnb, fgc)